src/mesh/high_order_grid.cpp

```diff
@@ -98,25 +102,33 @@ void HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::initialize_with
     reset_initial_nodes();
     if (output_mesh) output_results_vtk(nth_refinement++);
 
-    // Used to check Jacobian validity
-    const unsigned int exact_jacobian_order = (max_degree-1) * dim;
-    const unsigned int min_jacobian_order = 1;
-    const unsigned int used_jacobian_order = std::max(exact_jacobian_order, min_jacobian_order);
-    evaluate_lagrange_to_bernstein_operator(used_jacobian_order);
-
-    auto cell = dof_handler_grid.begin_active();
-    auto endcell = dof_handler_grid.end();
-    // pcout << "Disabled check_valid_cells. Took too much time due to shape_grad()." << std::endl;
-    for (; cell!=endcell; ++cell) {
-        if (!cell->is_locally_owned())  continue;
-        const bool is_invalid_cell = check_valid_cell(cell);
-
-        if ( !is_invalid_cell ) {
-            std::cout << " Poly: " << max_degree
-                      << " Grid: " << nth_refinement
-                      << " Cell: " << cell->active_cell_index() << " has an invalid Jacobian." << std::endl;
-            // bool fixed_invalid_cell = fix_invalid_cell(cell);
-            // if (fixed_invalid_cell) std::cout << "Fixed it." << std::endl;
+    if(check_valid_metric_Jacobian){
+        // Used to check Jacobian validity
+        //For future note: the 3D order of the Jacobian in each direction should be
+        // (max_degree-1) * max_degree * max_degree. Then exact_jacobian_order below
+        //is getting passed as a 1D order then a 3D finite element is created based off
+        // exact_jac_order*exact_jac_order*exact_jac_order which does not equal the above.
+        //Also, when the metric terms are built from operators: metric_operators
+        //it check the validity of the metric Jacobian on-the-fly.
+        const unsigned int exact_jacobian_order = (max_degree-1) * dim;
+        const unsigned int min_jacobian_order = 1;
+        const unsigned int used_jacobian_order = std::max(exact_jacobian_order, min_jacobian_order);
+        evaluate_lagrange_to_bernstein_operator(used_jacobian_order);
+         
+        auto cell = dof_handler_grid.begin_active();
+        auto endcell = dof_handler_grid.end();
+        // pcout << "Disabled check_valid_cells. Took too much time due to shape_grad()." << std::endl;
+        for (; cell!=endcell; ++cell) {
+            if (!cell->is_locally_owned())  continue;
+            const bool is_invalid_cell = check_valid_cell(cell);
+         
+            if ( !is_invalid_cell ) {
+                std::cout << " Poly: " << max_degree
+                          << " Grid: " << nth_refinement
+                          << " Cell: " << cell->active_cell_index() << " has an invalid Jacobian." << std::endl;
+                // bool fixed_invalid_cell = fix_invalid_cell(cell);
+                // if (fixed_invalid_cell) std::cout << "Fixed it." << std::endl;
```

Comment:
```suggestion
```

Reply:
This line isn't from this PR.

Reply:
Understood, resolving the ones you've indicated aren't from this PR.